File: 약수의_개수와_덧셈.py
def divisor(n):
    count = 1
    temp=1
    d=2
    while(n>1):
        if n%d == 0:
            n=n//d
            temp+=1
        else:
            count*=temp
            temp=1
            d+=1
    return count*temp
            

def solution(left, right):
    answer = 0
    for n in range(left, right+1):
        count = divisor(n)
        if count%2 == 0:
            answer+=n
        else:
            answer-=n
    
    return answer

File: java/Level 1/약수의_개수와_덧셈.java
class Solution {
    public int divisor(int n){
        int cnt = 1;
        int temp = 1;
        int d = 2;
        while(n > 1){
            if(n % d == 0){
                n = n / d;
                temp += 1;                    
            }
            else{
                cnt *= temp;
                temp = 1;
                d += 1;
            }
        }
        
        return cnt * temp;
    }
    
    
    public int solution(int left, int right) {
        int answer = 0;
        for(int i=left; i<=right; i++){
            int cnt = divisor(i);
            if(cnt % 2 == 0){
                answer += i;
            }
            else{
                answer -= i;
            }
        }
        return answer;
    }
}
